sqe-http-api/Controllers/ArtefactController.cs

```diff
@@ -78,33 +104,27 @@ public async Task<ActionResult<ArtefactListDTO>> GetArtefacts([FromRoute] uint e
         /// </summary>
         /// <param Name="editionId">Unique Id of the desired edition</param>
         [HttpPost("edition/{editionId}/artefact")]
-        public async Task<ActionResult<ArtefactListDTO>> CreateArtefact(
-            [FromRoute] uint editionId, 
-            [FromRoute] uint artefactId, 
+        public async Task<ActionResult<ArtefactDTO>> CreateArtefact(
+            [FromRoute] uint editionId,
             [FromBody] CreateArtefactDTO payload
         )
         {
             try
             {
-                return Ok(await _artefactService.CreateArtefact(
+                return await _artefactService.CreateArtefact(
                     _userService.GetCurrentUserObject(), 
                     editionId, 
                     payload.masterImageId, 
                     payload.mask, 
                     payload.name, 
-                    payload.position));
+                    payload.position);
             }
             catch (NotFoundException)
             {
                 return NotFound();
             }
         }
 
-        public class UpdateArtefactMask
-        {
-            public string mask { get; set; }
-        }
-
         public class UpdateArtefactDTO
         {
             public string mask { get; set; }
```

Comment:
Isn't it possible to update the Artefact's name, too?

Reply:
Yep, in the UpdateArtefactDTO you pass to the PUT request you can set any combination of the three: "mask", "name", and "position".